### PR TITLE
add models

### LINKED ISSUE #3: Data base models

### PR DESCRIPTION
**Pull request by AlizeeLfe**
Close #3

### CODE REVIEW

File: models/Sauce.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+const sauceSchema = mongoose.Schema({
+  name: { type: String, required: true },
+  manufacturer: { type: String, required: true },
+  description: { type: String, required: true },
+  mainPepper: { type: String, required: true },
+  imageUrl: { type: String, required: true },
+  heat: { type: Number, required: true },
+  likes: { type: Number, required: true },
+  dislikes: { type: Number, required: true },
+  usersLiked: { type: ["String <userId>"], required: true },
```
Comment (Tutt-J) [inline models/Sauce.js]: ["String <userId>"] not exist, it's juste represent the fact that you hav to link with user model.

For example :  usersLiked: [{ type: String, ref: "User" }],

File: models/Sauce.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// Importation de Mongoose
+const mongoose = require("mongoose");
+
+// Création d'un schéma de données (informations requises)
+const sauceSchema = mongoose.Schema({
+  name: { type: String, required: true },
```
Comment (Tutt-J) [inline models/Sauce.js]: Missing userId

File: models/User.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+const mongoose = require("mongoose");
+
+// Rajout d'un validateur comme plugin à notre schéma  (mongoose unique validator)
+//(pour pas avoir plusieurs utilisateurs avec la même adresse mail)
+// car "unique=true" inssufisant (possibles erreurs de Mongodb).
+const uniqueValidator = require("mongoose-unique-validator");
+
+// Création d'un schéma d'utilisateur
+const userSchema = mongoose.Schema({
+  email: { type: String, required: true, unique: true },
```
Comment (Tutt-J) [inline models/User.js]: Add error message to unique

**Review by Tutt-J — CHANGES_REQUESTED** [review]
And add some validation with validator like min length, mex length... And and trim

File: models/Sauce.js (hunk 2 above)
Comment (AlizeeLfe) [reply models/Sauce.js]: Maybe I misunderstand, but I learned that "userId" was automatically generated by MongoDB (database), and therefore it was not necessary to add it in the model. (that's why it was missing).

File: models/User.js (hunk 3 above)
Comment (AlizeeLfe) [reply models/User.js]: Do I need to add a "user.updateOne" and {runValidators: true} and a "then/catch" to update informations ?
I heard that Update validators are off by default.

File: models/Sauce.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+// Importation de Mongoose
+const mongoose = require("mongoose");
+
+// Création d'un schéma de données (informations requises)
+const sauceSchema = mongoose.Schema({
+  userId: { type: String, required: true, unique: true },
```
Comment (Tutt-J) [inline models/Sauce.js]: Missing ref User, not unique

**Review by Tutt-J — CHANGES_REQUESTED** [review]

**Review by Tutt-J — APPROVED** [review]

**Pull request merged**